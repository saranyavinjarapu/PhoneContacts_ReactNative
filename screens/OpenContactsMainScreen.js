import React from "react";
import { StyleSheet, View, Button, Alert } from "react-native";
import * as Permissions from "expo-permissions";
import * as Contacts from "expo-contacts";

const OpenContactsMainScreen = (props) => {
  const openContactsHandler = async () => {
    const { status, permissions } = await Permissions.askAsync(
      Permissions.CONTACTS
    );
    if (status === "granted") {
      const { data } = await Contacts.getContactsAsync({
        fields: [
          Contacts.Fields.Name,
          Contacts.Fields.PhoneNumbers,
          Contacts.Fields.Image,
        ],
        sort: Contacts.SortTypes.FirstName,
      });

      if (data.length > 0) {
        props.showContacts(data);
      } else {
        Alert.alert("No Contacts Found");
      }
    } else {
      Alert.alert("Permission to access contacts denied");
    }
  };
  return (
    <View style={styles.screen}>
      <View style={styles.buttonContainer}>
        <Button
          textStyle={{ color: "#FFFFFF", fontSize: 400 }}
          title="OPEN PHONE CONTACTS"
          onPress={openContactsHandler}
          style={styles.openContactButton}
        ></Button>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    padding: 50,
  },
  buttonContainer: {
    flexDirection: "row",
    alignItems: "center",
    width: "100%",
    justifyContent: "center",
    height: 200,
    shadowColor: "black",
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 20,
    shadowOpacity: 0.26,
    backgroundColor: "white",
    elevation: 15,
    padding: 20,
    borderRadius: 50,
    marginTop: 100,
  },
  openContactButton: {
    borderColor: "black",
    borderWidth: 5,
  },
});

export default OpenContactsMainScreen;
