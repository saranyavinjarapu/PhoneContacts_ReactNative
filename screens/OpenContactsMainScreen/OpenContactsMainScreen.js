import React from "react";
import { View, Button, Alert } from "react-native";
import styles from "./OpenContactsMainScreenStyles";
import * as Permissions from "expo-permissions";
import * as Contacts from "expo-contacts";

const OpenContactsMainScreen = (props) => {
  const openContactsHandler = async () => {
    const { status, permissions } = await Permissions.askAsync(
      Permissions.CONTACTS
    );
    if (status === "granted") {
      const { data } = await Contacts.getContactsAsync({
        fields: [
          Contacts.Fields.Name,
          Contacts.Fields.PhoneNumbers,
          Contacts.Fields.Image,
        ],
        sort: Contacts.SortTypes.FirstName,
      });

      if (data.length > 0) {
        props.showContacts(data);
      } else {
        Alert.alert("No Contacts Found");
      }
    } else {
      Alert.alert("Permission to access contacts denied");
    }
  };
  return (
    <View style={styles.screen}>
      <View style={styles.buttonContainer}>
        <Button
          textStyle={{ color: "#FFFFFF", fontSize: 400 }}
          title="OPEN PHONE CONTACTS"
          onPress={openContactsHandler}
          style={styles.openContactButton}
        ></Button>
      </View>
    </View>
  );
};

export default OpenContactsMainScreen;
