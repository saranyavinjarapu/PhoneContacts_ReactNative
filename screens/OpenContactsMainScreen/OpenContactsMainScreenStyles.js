import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    padding: 50,
  },
  buttonContainer: {
    flexDirection: "row",
    alignItems: "center",
    width: "100%",
    justifyContent: "center",
    height: 200,
    shadowColor: "black",
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 20,
    shadowOpacity: 0.26,
    backgroundColor: "white",
    elevation: 15,
    padding: 20,
    borderRadius: 50,
    marginTop: 100,
  },
  openContactButton: {
    borderColor: "black",
    borderWidth: 5,
  },
});

export default styles;
