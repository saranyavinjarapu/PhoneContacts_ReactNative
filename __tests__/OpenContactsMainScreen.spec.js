import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import OpenContactsMainScreen from "../screens/OpenContactsMainScreen/OpenContactsMainScreen";

describe("Open Contacts Main Screen Renders Correctly", () => {
  it("renders correctly", () => {
    const snap = render(<OpenContactsMainScreen />).toJSON();
    expect(snap).toMatchSnapshot();
  });

  it("Open Contacts Button Works", () => {
    const { queryByText } = render(<OpenContactsMainScreen />);
    const sendHandler = jest.fn();
    const button = queryByText("OPEN PHONE CONTACTS");
    expect(button).toBeTruthy();
    fireEvent.press(button);
    expect(sendHandler).not.toBeNull();
  });
});
