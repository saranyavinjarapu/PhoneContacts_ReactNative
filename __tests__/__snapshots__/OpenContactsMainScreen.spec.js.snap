// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Open Contacts Main Screen Renders Correctly renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 50,
      "paddingTop": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 50,
        "elevation": 15,
        "flexDirection": "row",
        "height": 200,
        "justifyContent": "center",
        "marginTop": 100,
        "padding": 20,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.26,
        "shadowRadius": 20,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          OPEN PHONE CONTACTS
        </Text>
      </View>
    </View>
  </View>
</View>
`;
